akka-stream/src/main/scala/akka/stream/impl/ActorProcessor.scala

```diff
@@ -174,7 +184,14 @@ private[akka] class SimpleOutputs(val actor: ActorRef, val pump: Pump) extends D
     }
   }
 
-  def cancel(e: Throwable): Unit = {
+  override def cancel(): Unit = {
+    if (!downstreamCompleted) {
+      downstreamCompleted = true
+      if (exposedPublisher ne null) exposedPublisher.shutdown(None)
+    }
+  }
+
+  override def error(e: Throwable): Unit = {
```

Comment:
well, I consider it to be on the reactive stream level since it is a command to call tryOnError